Showing changes made from my dumb early push to master

cjacobwade opened the pull request:
Adding working resource bar, collecting resources and feeding buddies. Improved way of showing resources/beacons so we can reuse the resource prefabs. Added dummy buddy to test feeding/cuteness. Adding several models, materials, etc from the art jam.

cjacobwade committed:
Resource gathering/giving to buddies

cjacobwade committed:
Merge commit '2d771b6405817d3310e09b54e69e6571d9373bdb' into Chris-Dev

cjacobwade committed:
Changing prefab values

cjacobwade committed:
Adding affect to jump when feeding buddies, adding #if editor tags to get the game building

cjacobwade committed:
Merge branch 'master' into Chris-Dev

brickman1444 committed:
Reorganizing some scripts. Add unityvs to gitignore (again?)

cjacobwade committed:
Merge commit 'a519c3643d7f6d6188823cd90f6a9b39c747cc0c' into Chris-Dev

Conflicts:
	Assets/Environment/SmallMountains.asset
	Assets/Scripts/Avatar.cs

cjacobwade committed:
Fixing weird offset terrain

brickman1444 committed:
Merge branch 'master' into organizing_files2

Conflicts:
	Assets/Scenes/Garden.unity
	Assets/Scripts/Resources/resource.cs

randomPoison commented on Assets/Scripts/Actors/ActorComponents/ActorResources.cs:
```diff
@@ -43,12 +48,21 @@ void Start ()
+			
 			int numResources = heldResourceTypes.Count;
-
+			
```
> ActorResources.cs:65

You've got trailing whitespace.

randomPoison commented on Assets/Scripts/Actors/ActorComponents/ActorResources.cs:
```diff
@@ -59,23 +73,69 @@ void Update ()
 			}
 			
-			resourceIndex = Mathf.FloorToInt(Mathf.Clamp(nextIndex, 0, heldResources.Count - 1));
+			resourceIndex = Mathf.FloorToInt(Mathf.Clamp(nextIndex, 0, heldResourceTypes.Count - 1));
```
> ActorResources.cs:76

`Mathf.Clamp()` is overloaded for `int`, you don't need to do `FloorToInt()`.

randomPoison commented on Assets/Scripts/Actors/ActorComponents/ActorResources.cs:
```diff
@@ -94,4 +154,19 @@ void SpawnResourceObject()
+		{
+			resourceComponent.used = true;
+
+			Debug.Log(other.gameObject.name);
```
> ActorResources.cs:165

I'd like to limit what debug logging goes into master. Things that give generally useful output are okay, but things that only provide information used in developing one subsystem should be removed before doing a pull request. I think this falls into the latter category.

cjacobwade replied:
Sounds good.

brickman1444 committed:
Fixing script loading errors

brickman1444 committed:
Capitalizing file names

brickman1444 committed:
Capitalizing Resource class name

brickman1444 committed:
Merge pull request #9 from Capstrong/organizing_files2

Organizing Files

randomPoison committed:
Buddies Follow Nearby Gods
- Buddies now check if there are any gods nearby and follow them around in addition to wandering around idly.
- Split shared node info into BehaviorTreeInfo class.
- Made some previously god- or buddy-specific nodes into general purpose nodes.
- Made separate nodes for moving to a Vec3 destination and following a Transform target.

randomPoison committed:
Remove Existing Behavior Tree
- Removing the behavior tree code in preparation of making it a submodule.

randomPoison committed:
Add Behavior Tree As Submodule

randomPoison committed:
Make Buddies Actors
- Added Actor and ActorPhysics sripts to the Buddy prefab to make use of ActorPhysics for movement.
- Renamed MoveToTarget to FollowTarget and updated assets as appropriate.
- Made changes to the Actor systems to allow them to be used on the buddies.
- Moved common nodes into GeneralNodes.cs

randomPoison committed:
Rework Actor Systems
- Removed combat and spell code since combat is no longer part of the game.
- Added PlayerActor and PlayerActorPhysics and moved player-specific code into them.
- Recreated the Player prefab from the player in the current scene.
- Cleanup of warnings in the Actor code.

heero6528 committed:
Added Greek Ruins

heero6528 committed:
New Texture and Grey texture

randomPoison committed:
Remove Reference to ActorResources from Actor.cs
- This was meant to be part of the previous commit but I forgot :(

randomPoison committed:
Merge branch 'master' into lildude_behaviors

Conflicts:
	Assets/Scripts/BehaviorTree
	Assets/Terrain.asset

randomPoison committed:
Rename ActorResources to PlayerActorResources
- Remove GodInfo and replaced it with GodTag.
- Continued to convert getters in Actor to properties.

randomPoison committed:
Rename LildudeNodes.cs to BuddyNodes.cs
- Removed GodNodes.cs and God Behavior.asset since they are no longer in use (and can easily be remade later).

randomPoison committed:
Add Clang Format Config File
- Formated files according to the config file.

randomPoison committed:
Buddies Face Direction They're Walking
- Moved ModelControl() back into ActorPhysics and have MoveAtSpeed() call it automatically.

randomPoison committed:
Buddy Test Animation
- Added a simply walk animation to make the buddies seem move lively.

randomPoison committed:
Merge branch 'lildude_behaviors' into format_test

Conflicts:
	Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs
	Assets/Scripts/Actors/ActorComponents/PlayerActorPhysics.cs

randomPoison committed:
Merge pull request #11 from Capstrong/format_test

Formatting Configuration

randomPoison committed:
Code Review Cleanup
- Removed ActorPhysics.IsGrabbing() since it's not used.
- Added a reference to the PlayerActorResources to PlayerActor.

randomPoison committed:
Merge pull request #10 from Capstrong/lildude_behaviors

Simple Buddy Behaviors and Actor Refactoring

randomPoison commented:
I think this is done, so I'm closing it down.
